x-pack/plugins/cloud_security_posture/common/types.ts

```diff
@@ -47,8 +47,14 @@ export interface ComplianceDashboardData {
   trend: PostureTrend[];
 }
 
+export type Status = 'indexed' | 'indexing' | 'index timeout' | 'not deployed' | 'not installed';
+
 export interface CspSetupStatus {
-  latestFindingsIndexStatus: 'applicable' | 'inapplicable';
+  status: Status;
+  installed_pkg_ver?: string;
+  latest_pkg_ver: string;
+  installed_integration: number;
+  healthy_agents: number;
```

Comment:
whenever `status` is `not_installed`, the `installed_pkg_ver` property shouldn't exists, and this interface allows for it to be either `undefined` or a `string`, which is incorrect. even if our code doesn't generate this, the types allow it.

this can be easily fixed by making `CspSetupStatus` a [discriminated union](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions) type where each interface declares its `status`  and the rest of it's relevant properties. by doing this we'll we make sure there are no invalid states, which we always get once we mix interfaces and use `optional` properties to hide it.